src/main/java/teammates/common/util/ActivityLogGenerator.java

```diff
@@ -0,0 +1,336 @@
+                        + "Show/Hide Details >>"
+                    + "</a>"
+                    + "<br>"
+                    + "<span id=\"error" + errorEmail.hashCode() + "\" style=\"display: none;\">"
+                        + errorEmail.getContent()
+                    + "</span>";
+            builder.withLogMessage(message);
+        }
+        
+        builder.withActionName(Const.ACTION_RESULT_SYSTEM_ERROR_REPORT);
+        // TODO role is unclear
+       
+        return builder.build().generateLogMessage();
+    }
+    
+    private ActivityLogEntry.Builder generateBasicActivityLogEntry(HttpServletRequest req) {
+        String url = HttpRequestHelper.getRequestedUrl(req);
+        String servletName = getActionNameFromUrl(url);
+        long currTime = System.currentTimeMillis();
+        ActivityLogEntry.Builder builder = new ActivityLogEntry.Builder(servletName, url, currTime);
+        builder.withActionName(servletName); // by default
+        
+        String logId;
+        if (isAutomatedAction(url)) {
+            logId = generateLogIdInAutomatedAction(currTime);
+            builder.withUserRole(Const.ActivityLog.ROLE_AUTO);
+        } else if (isUserGoogleLogIn()) {
+            logId = generateLogIdWithGoogleId(getGoogleId(), currTime);
+            builder.withUserGoogleId(getGoogleId());
+        } else {
+            logId = generateLogIdWithoutGoogleId(req, currTime);
+            builder.withUserGoogleId(Const.ActivityLog.AUTH_UNLOGIN);
+        }
+        builder.withLogId(logId);
+        
+        return builder;
+    }
+    
+    /**
+     * Generates log message for automated action and public source accessing action.
+     * <p>
+     * If the user has logged in with google id or authenticated key, his/her identity will be recorded
+     * in the log id. If the action is automated action, 'Auto' will be used instead for the identity.
+     * 
+     * @param req HttpServletRequest in the request
+     * @param message log message to show to admin
+     * @return log message in form specified in {@link ActivityLogEntry}
+     */
+    public String generateBasicActivityLogMessage(HttpServletRequest req, String message) {
+        ActivityLogEntry.Builder builder = generateBasicActivityLogEntry(req);
+        
+        builder.withLogMessage(message);
+        
+        return builder.build().generateLogMessage();
+    }
+    
+    private boolean isAutomatedAction(String url) {
+        return url.startsWith(Const.ActivityLog.PREFIX_AUTO_PAGE);
+    }
+    
+    private boolean isUserGoogleLogIn() {
+        return userService.getCurrentUser() != null;
+    }
+
+    private String getGoogleId() {
+        if (!isUserGoogleLogIn()) {
+            return Const.ActivityLog.AUTH_UNLOGIN;
+        }
+        return userService.getCurrentUser().getNickname();
+    }
+
+    private String getActionNameFromUrl(String requestUrl) {
```

Comment:
Thanks for suggestion, but here the action name means the thing behind "page(admin)" in url (e.g `page/myActionName` `admin/myActionName`)